static/src/javascripts/bootstraps/common.js

```diff
@@ -384,8 +392,8 @@ define([
             },
 
             initDiscussion: function () {
-                mediator.on('page:common:ready', function () {
-                    if (config.switches.discussion) {
+                if (config.switches.discussion) {
+                    mediator.on('page:common:ready', function () {
```

Comment:
NB. moved this (& facebook above) because it seems best to attach a minimum of event listeners. So as a pattern, do it conditionally on a switch, not the other way round.